lib/src/rules/prefer_on_push_cd_strategy.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:dart_code_metrics/src/models/code_issue.dart';
+import 'package:dart_code_metrics/src/models/code_issue_severity.dart';
+
+import 'base_rule.dart';
+import 'rule_utils.dart';
+
+class PreferOnPushCdStrategyRule extends BaseRule {
+  static const String ruleId = 'prefer-on-push-cd-strategy';
+
+  static const _failure = 'Prefer using onPush change detection strategy.';
+
+  PreferOnPushCdStrategyRule({Map<String, Object> config = const {}})
+      : super(
+          id: ruleId,
+          severity: CodeIssueSeverity.fromJson(config['severity'] as String) ??
+              CodeIssueSeverity.warning,
+        );
+
+  @override
+  Iterable<CodeIssue> check(
+      CompilationUnit unit, Uri sourceUrl, String sourceContent) {
+    final visitor = _Visitor();
+
+    unit.visitChildren(visitor);
+
+    return visitor.expression
+        .map((expression) => createIssue(this, _failure, null, null, sourceUrl,
+            sourceContent, unit.lineInfo, expression))
+        .toList(growable: false);
+  }
+}
+
+class _Visitor extends RecursiveAstVisitor<Object> {
+  final _expression = <AstNode>[];
+
+  Iterable<AstNode> get expression => _expression;
+
+  @override
+  void visitAnnotation(Annotation node) {
+    if (!_isComponentAnnotation(node)) {
+      return;
+    }
+
+    final changeDetectionArg = node.arguments.arguments
+        .whereType<NamedExpression>()
+        .firstWhere((arg) => arg.name.label.name == 'changeDetection',
+            orElse: () => null);
+
+    if (changeDetectionArg == null) {
+      return _expression.add(node);
+    }
+
+    final strategyValue = changeDetectionArg.expression;
+    if (strategyValue is! PrefixedIdentifier) {
+      return _expression.add(changeDetectionArg);
+    }
+
+    if (_isStrategyValueCorrect(strategyValue as PrefixedIdentifier)) {
+      return;
+    }
+
+    return _expression.add(changeDetectionArg);
```

Comment:
It seems that you can simplify check and get rid of lines 56-58
```dart
    if (strategyValue is PrefixedIdentifier && _isStrategyValueCorrect(strategyValue)  {
       return; 
    }
    
    return _expression.add(changeDetectionArg);
```

Reply:
indeed, thank you! Fixed.

Reply:
I guess you should just `return` instead of `return _expression.add(changeDetectionArg);` on [line 57](https://github.com/wrike/dart-code-metrics/pull/85/files#diff-059454b75e7f06023b6f4ceed0431748R57)

Reply:
Yes, tests are failed. Fixed it. My bad